Icons: Update material icons

QianMoXi opened the pull request:
## Description
As a follow-up to https://github.com/MudBlazor/MudBlazor/pull/6023, update material icon to the latest
However, there are 3 icons that have been removed:

- PhoneInTalk: no record
- Flourescent: renamed to Fluorescent, related [material-design-icons #1208](https://github.com/google/material-design-icons/issues/1208)
- Whatsapp: legal reasons, related [material-design-icons #1366](https://github.com/google/material-design-icons/issues/1366)

## How Has This Been Tested?
visually

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] The PR is submitted to the correct branch (`dev`).
- [x] My code follows the code style of this project.
- [ ] I've added relevant tests.

QianMoXi committed:
Update material icons to the latest

mikes-gh commented:
Probably we need to Forward `Flourescent` to the new spelling
WhatsApp can be forwarded in to `Icons.Custom.Brands`
The misssing one can also go in `_Obsolete.cs` too.
We should also update the forwarded Icons in this file to there full namespace versions.
It would be nice to remove everything else in the obsolete folder eventually so removing any library dependencies to the old non qualified Icons would be a good exercise.
I think it can all go in this PR

mikes-gh commented:
I also noticed the script is adding to the Obsolete Folder for new Icons.  We shouldn't do that

mikes-gh commented:
@ScarletKuro I presume we can remove the Obsolete folders except `_Obsolete.cs` (prob major version) without breaking the trimming?

ScarletKuro commented:
@mikes-gh yes, this is only for [compatibility purpose](https://github.com/MudBlazor/MudBlazor/pull/4639https://github.com/MudBlazor/MudBlazor/pull/4639)(read the nota bene). There is a task in project https://github.com/MudBlazor/MudBlazor/projects/13#card-82213304
You will need to modify the script and change places that using the obsolete path to icons.

QianMoXi committed:
Remove new icons in Obsolete folder, add outdated icons to _Obsolete.cs

QianMoXi committed:
Remove the part of the script that made changes to the Obsolete folder

QianMoXi committed:
Update IconCodeOutput in the document

QianMoXi commented:
> Forwards can go here
> 
> https://github.com/MudBlazor/MudBlazor/blob/dev/src/MudBlazor/Icons/Obsolete/_Obsolete.cs
> 
> Probably we need to Forward `Flourescent` to the new spelling WhatsApp can be forwarded in to `Icons.Custom.Brands.WhatsApp` The misssing one can also go in `_Obsolete.cs` too. We should also update the forwarded Icons in this file to their fully qualified versions. It would be nice to remove everything else in the obsolete folder eventually so removing any library dependencies to the old non qualified Icons would be a good exercise. I think it can all go in this PR

I've moved the icon to _Obsolete.cs

Whatsapp is visually different from WhatsApp, so I didn't forward to Icons.Custom.Brands.WhatsApp
![image](https://user-images.githubusercontent.com/26250460/209502227-0c18dc93-7abc-4831-8cb2-c479f0e9155a.png)

> I also noticed the script is adding to the Obsolete Folder for new Icons. We shouldn't do that

I thought it was designed this way.
Now that I've modified the script, it doesn't change the Obsolete folder anymore

At the same time, I modified the `IconCodeOutput` in the docs to match the new icon
![image](https://user-images.githubusercontent.com/26250460/209502803-840cc27e-0596-446e-afc9-c4001d8e8a5f.png)

mikes-gh commented:
@QianMoXi Sorry to mess you around butt regarding the Obsolete folder (as er @ScarletKuro comment) keep the adding to obsolete folder as its more obsolete path access rather than individual icons.  
We will drop the folder in the next major version.

Then we can merge.  The rest looks good.  Thanks a lot.

QianMoXi committed:
Revert any changes to Obsolete folder

QianMoXi commented:
@mikes-gh Sorry, I didn't fully understand what you mean yesterday because English is not my native language.

I've fixed my commit, you can check it out

Should we explicitly add `ObsoleteAttribute` in Obsolete folder(e.g `Icons.Filled`)? In this case, we can remind the user that they should switch to the `Icons.Material` class.

mikes-gh commented:
@QianMoXi Sorry for not being clear .  Your last commit is not what I meant.  I meant to change the script back to write new icons to obsolete folders on generation.
I think there are also some valid changes in the last commit as well though.
We are going to keep a copy of all the icons as Short unqualified names until we remove them all.

Does that make sense?

mikes-gh commented:
> Should we explicitly add `ObsoleteAttribute` in Obsolete folder(e.g `Icons.Filled`)? In this case, we can remind the user that they should switch to the `Icons.Material` class.
Yes that sounds like a good idea.

QianMoXi committed:
Restore the script and execute it again

QianMoXi committed:
add ObsoleteAttribute in Icons Obsolete folder

QianMoXi committed:
Update the code for the icon to remove all build warnings

QianMoXi commented:
@mikes-gh I'm done.

> Should we explicitly add ObsoleteAttribute in Obsolete folder(e.g Icons.Filled)? 

I replaced the access icon path with a regular expression. 
Others might need it too, so I posted it

original:
```
Icons.(Filled|Rounded|Outlined|Sharp|TwoTone).
```

replace:
```
Icons.Material.$1.
```

mikes-gh committed:
Minor script style updates

mikes-gh commented:
@QianMoXi thanks a lot for this. LGTM
@ScarletKuro can you have a quick look please 🙏

ScarletKuro commented:
@mikes-gh LGTM
